packages/mgt-components/src/components/mgt-file-list/mgt-file-list.ts

```diff
@@ -0,0 +1,592 @@
+   *
+   * @type {string}
+   * @memberof MgtFile
+   */
+  @property({
+    attribute: 'user-id'
+  })
+  public get userId(): string {
+    return this._userId;
+  }
+  public set userId(value: string) {
+    if (value === this._userId) {
+      return;
+    }
+
+    this._userId = value;
+    this.requestStateUpdate();
+  }
+
+  /**
+   * allows developer to provide insight type for a file
+   * can be trending, used, or shared
+   *
+   * @type {OfficeGraphInsightString}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'insight-type'
+  })
+  public get insightType(): OfficeGraphInsightString {
+    return this._insightType;
+  }
+  public set insightType(value: OfficeGraphInsightString) {
+    if (value === this._insightType) {
+      return;
+    }
+
+    this._insightType = value;
+    this.requestStateUpdate();
+  }
+
+  /**
+   * Sets what data to be rendered (file icon only, oneLine, twoLines threeLines).
+   * Default is 'threeLines'.
+   *
+   * @type {ViewType}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'view',
+    converter: value => {
+      if (!value || value.length === 0) {
+        return ViewType.threelines;
+      }
+
+      value = value.toLowerCase();
+
+      if (typeof ViewType[value] === 'undefined') {
+        return ViewType.threelines;
+      } else {
+        return ViewType[value];
+      }
+    }
+  })
+  public view: ViewType;
+
+  /**
+   * allows developer to provide file type to filter the list
+   * can be docx
+   *
+   * @type {string[]}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'file-extensions',
+    converter: (value, type) => {
+      return value.split(',').map(v => v.trim());
+    }
+  })
+  public get fileExtensions(): string[] {
+    return this._fileExtensions;
+  }
+  public set fileExtensions(value: string[]) {
+    if (arraysAreEqual(this._fileExtensions, value)) {
+      return;
+    }
+
+    this._fileExtensions = value;
+    this.requestStateUpdate();
+  }
+
+  /**
+   * A number value to indicate the maximum number of files to show
+   * @type {number}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'show-max',
+    type: Number
+  })
+  public showMax: number;
+
+  /**
+   * A number value to indicate the number of more files to load when show more button is clicked
+   * @type {number}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'load-more-file-count',
+    type: Number
+  })
+  public loadMoreFileCount: number;
+
+  private _fileListQuery: string;
+  private _fileQueries: string[];
+  private _files: MicrosoftGraph.DriveItem[];
+  private _siteId: string;
+  private _itemId: string;
+  private _driveId: string;
+  private _itemPath: string;
+  private _groupId: string;
+  private _insightType: OfficeGraphInsightString;
+  private _fileExtensions: string[];
+  private _userId: string;
+  private _renderedFileCount: number;
+  private _filesToRender: MicrosoftGraph.DriveItem[];
```

Comment:
You can make this an @internalProperty and avoid having to call requestUpdate - when the property changes, it will rerender

```suggestion
  @internalProperty() private _filesToRender: MicrosoftGraph.DriveItem[];
```